lms/lmsweb/views.py

```diff
@@ -208,6 +208,50 @@ def share():
     return fail(400, f'Unknown or unset act value "{act}".')
 
 
+@webapp.route('/notes/<int:user_id>', methods=['GET', 'POST'])
```

Comment:
It's a bit hard to read this complicate function. I'd probably try to send some of the code to models/notes.py